airflow/migrations/env.py

```diff
@@ -98,7 +98,23 @@ def run_migrations_online():
         )
 
         with context.begin_transaction():
+            if connection.dialect.name == 'mysql' and \
+                    connection.dialect.server_version_info >= (5, 6):
+                connection.execute(
+                    "select GET_LOCK('alembic',1800);"
+                )
+            if connection.dialect.name == 'postgresql':
+                context._ensure_version_table()  # pylint: disable=protected-access
```

Comment:
```suggestion
                context.get_context()._ensure_version_table()  # pylint: disable=protected-access
```

Because the `_ensure_version_table` function is only available in the `MigrationContext` you have to get this context from the (environment) context you are in.

From the `MigrationContext` docs:
```
    When inside the ``env.py`` script, the :class:`.MigrationContext`
    is available via the
    :meth:`.EnvironmentContext.get_context` method,
    which is available at ``alembic.context``::

        # from within env.py script
        from alembic import context
        migration_context = context.get_context()
```

Reply:
Actually I think `run_migrations` will take care of this:
```python
...
        if self.purge:
            if self.as_sql:
                raise util.CommandError("Can't use --purge with --sql mode")
            self._ensure_version_table(purge=True)
            heads = ()
        else:
            heads = self.get_current_heads()

            if not self.as_sql and not heads:
                self._ensure_version_table()
...
```
Which case is missing @anitakar?